tests/e2e/Services/Functions/FunctionsConsoleClientTest.php

```diff
@@ -410,4 +448,52 @@ public function testDeleteVariable(array $data)
 
         return $data;
     }
+
+    public function testVariableE2E(): void
+    {
+        $function = $this->createFunction([
+            'functionId' => ID::unique(),
+            'runtime' => 'node-18.0',
+            'name' => 'Variable E2E Test',
+            'entrypoint' => 'index.js',
+            'logging' => false,
+            'execute' => ['any']
+        ]);
+
+        $this->assertEquals(201, $function['headers']['status-code']);
+        $this->assertFalse($function['body']['logging']);
+        $this->assertNotEmpty($function['body']['$id']);
+
+        $functionId = $functionId = $function['body']['$id'] ?? '';
+
+        // create variable
+        $variable = $this->createVariable($functionId, [
+            'key' => 'CUSTOM_VARIABLE',
+            'value' => 'test',
+            'secret' => true,
+        ]);
+
+        $this->assertEquals(201, $variable['headers']['status-code']);
+        $this->assertNotEmpty($variable['body']['$id']);
+        $this->assertEquals('CUSTOM_VARIABLE', $variable['body']['key']);
+        $this->assertEquals('', $variable['body']['value']);
+        $this->assertEquals(true, $variable['body']['secret']);
+
+        $deploymentId = $this->setupDeployment($functionId, [
+            'entrypoint' => 'index.js',
+            'code' => $this->packageFunction('node'),
+            'activate' => true
+        ]);
+
+        $this->assertNotEmpty($deploymentId);
+
+        $execution = $this->createExecution($functionId);
+
+        $this->assertEquals(201, $execution['headers']['status-code']);
+        $this->assertEmpty($execution['body']['logs']);
+        $this->assertEmpty($execution['body']['errors']);
+        $this->assertStringContainsString('"CUSTOM_VARIABLE":"test"', $execution['body']['responseBody']);
```

Comment:
Instead of this, since response is JSON, let's do something like

```
$body = json_decode(execution.body.responseBody)
```
So then the test can be along lines
```
assertEquals('aSecretValue', $body['CUSTOM_VARIABLE'])
```